php/tests/generated_class_test.php

```diff
@@ -226,6 +227,37 @@ public function testEnumField()
         // Set string.
         $m->setOptionalEnum("1");
         $this->assertEquals(TestEnum::ONE, $m->getOptionalEnum());
+
+        // Test Enum methods
+        $this->assertEquals('ONE', TestEnum::name(1));
+        $this->assertEquals(1, TestEnum::value('ONE'));
+    }
+
+    /**
+     * @expectedException LogicException
+     * @expectedExceptionMessage Classes implementing Google\Protobuf\EnumTrait must define the static property $valueToName to use this function
+     */
+    public function testBadEnumThrowsException()
```

Comment:
why this test is needed? ```valueToName``` is implementation detail. That should not be part of API

Reply:
I just wanted to test `EnumTrait` responds in a sane way if implemented incorrectly. Since there's no way to prevent users from implementing it themselves. I can remove the test if you'd like.

Reply:
If you do need it, you can put it in the php_implementation_test

Reply:
Done!